src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py

```diff
@@ -57,9 +58,19 @@ def battery_inverter_mapping(batteries: Iterable[int]) -> dict[int, int]:
     return bat_inv_map
 
 
+@dataclass
+class TemperatureMetrics:
+    """Container for temperature metrics."""
+
+    timestamp: datetime
+    max: Temperature
+    min: Temperature
+    avg: Temperature
```

Comment:
I think this is too specific and breaking `Sample` as the main type used by formula engines. If we go this route I would much rather see a generic `SampleRange` or something like that, that provides the `max`,  `min` and `avg` similar to `Sample3Phase`. But them I'm not sure if then this can be easily composed using formulas, as probably formulas expect only one value per sample. The more I think about this approach of packed samples the more I think it is not a good idea. When we designed the pipeline we specifically agreed to make each metric a single value, otherwise we could have packed all metrics of a single component as one single sample, but there were issues with that (which I can't remember right now, although formula composition seems to be a good one).

But I guess @shsms should clarify this as he is the author of both `Sample3Phase` and the formula engine.

Reply:
So what i could do instead is unfold the idea and expose a `BatteryPool.temperature_max` along with `...temperature_min` and `...temperature_avg` receivers. This way we do not feed the concern you're raising. I'm fine with that.

Reply:
p.s.: that means i can refactor back to `Sample[Tenmperature]` that way then, too, if you agree.

Reply:
This won't work,  because of the throttling provided by `SendOnUpdate`.  Temperatures for some components would get dropped.  It might be better to use something like this:

``` python
class BatteryTemperatures:
    @dataclass
    class Metrics:
        component_id: int
        lowest: Temperature
        """Temperature of block with the lowest temperature."""
        highest: Temperature
        """Temperature of block with the highest temperature."""
        average: Temperature
        """Average temperature of all blocks."""

    def __init__(self, values: list[BatteryTemperatures.Metrics]):
        self._values = values

    def __iter__(self) -> Iterator[BatteryTemperatures.Metrics]:
        return iter(self._values)
```

also it was suggested that calling them `max` and `min` was confusing, because they sound like bounds.  `highest` and `lowest` might be better.

Reply:
@llucax formulas can only be applied on resampled values.  For a generic SampleRange with highest/lowest, etc.  I don't mind,  if you think there will be additional use cases for these in the future.

Reply:
> p.s.: that means i can refactor back to Sample[Tenmperature] that way then, too, if you agree.

that won't work,  because we have to stream temperatures of individual batteries.  Not an aggregate temperature.

Reply:
> @llucax formulas can only be applied on resampled values. For a generic SampleRange with highest/lowest, etc. I don't mind, if you think there will be additional use cases for these in the future.

Moving to https://github.com/frequenz-floss/frequenz-sdk-python/discussions/494

I'm fine with finishing this PR with an intermediary solution, but I think the discussion is worth having, specially with v1.0 approaching.